### PR TITLE
MULE-19485: Error is not correctly handled by Source Policy when execute-next is within a subflow

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

**Comment by mbuchwald** [comment]
--validate

File: core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java [hunk 1]
```diff
@@ -103,30 +108,48 @@ public CoreEvent process(CoreEvent event) throws MuleException {
   private OnExecuteNextErrorConsumer errorConsumer(PolicyEventMapper policyEventMapper,
                                                    PolicyNotificationHelper notificationHelper) {
 
-    if (isWithinSourcePolicy(getLocation())) {
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java:106]: if you remove this pre-check you will add complexity to every error handling, affecting perfromance

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java [hunk 2]
```diff
@@ -103,30 +108,48 @@ public CoreEvent process(CoreEvent event) throws MuleException {
...
+    List<FlowStackElement> elements = me.getEvent().getFlowCallStack().getElements();
+    if (elements.size() != 1) {
+      return false;
+    }
+
+    Location completeLocation = builderFromStringRepresentation(elements.get(0).getProcessorPath().split(" ")[0]).build();
+    List<String> parts = completeLocation.getParts();
+    boolean isSubflowWithinASource = completeLocation.getGlobalName().equals(getRootContainerLocation().toString())
+        && parts.size() >= 1 && parts.get(0).equals(SOURCE_POLICY_PART_IDENTIFIER);
+    return isSubflowWithinASource;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java:152]: return directly, without aux var

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java (hunk 2 above)
Comment (mbuchwald) [reply core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java:152]: I added the variable just for legibility. I can remove it, tho

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**